pkg/registry/daemonset/strategy.go

```diff
@@ -99,6 +99,14 @@ func (daemonSetStrategy) ValidateUpdate(ctx api.Context, obj, old runtime.Object
 	return append(validationErrorList, updateErrorList...)
 }
 
+func (daemonSetStrategy) SupportGraceful() bool {
```

Comment:
We need a runtime.Typer to get the ObjectMeta of a runtime.Object. I'm plumbing it throuhg DeleteStrategy.

Regarding why I added SupportGraceful(), it's because previously the BeforeDelete() quickly returns if the DeleteStrategy is nil, but now I force DeleteStrategy to be non-nil, so we need to add SupportGraceful() to return quickly.